morango/api/viewsets.py

```diff
@@ -292,26 +303,60 @@ def create(self, request):
 
         return response.Response(resp_data, status=status.HTTP_201_CREATED)
 
+    def retrieve(self, request, *args, **kwargs):
+        nonce = request.query_params.get("nonce", None)
+        signature = request.query_params.get("signature", None)
+
+        if not nonce or not signature:
+            return response.Response(
+                "Requests to get a sync session must have signature data",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        sync_session = self.get_object()
+
+        message = "{nonce}:{id}".format(nonce=nonce, id=sync_session.id)
+        if not sync_session.client_certificate.verify(message, signature):
+            return response.Response(
+                "Client certificate failed to verify signature",
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+        # check that the nonce is valid, and consume it so it can't be used again
+        try:
+            certificates.Nonce.use_nonce(nonce)
+        except errors.MorangoNonceError:
+            return response.Response(
+                "Nonce is not valid", status=status.HTTP_403_FORBIDDEN
+            )
+
+        return response.Response(self.get_serializer(sync_session).data)
+
```

Comment:
Yeah I went this route out of an abundance of caution, since it wasn't possible to GET a sync session (and its data) prior to this, and if any data was returned from the server regarding a sync session, it was only during create which required validating a nonce.

If you don't think we need it, I can remove the nonce handling. Or perhaps we make it optional through configuration?